diff --git a/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp b/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
index 5ee5d6ae2d..ad1b0382a5 100644
--- a/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
+++ b/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
@@ -243,12 +243,9 @@ nsresult MediaEngineRemoteVideoSource::Allocate(
   webrtc::CaptureCapability newCapability;
   LOG("ChooseCapability(%s) for mCapability (Allocate) ++",
       ToString(distanceMode));
-  if (!ChooseCapability(c, aPrefs, newCapability, distanceMode,
-                        aOutBadConstraint)) {
-    if (aOutBadConstraint && !*aOutBadConstraint) {
-      *aOutBadConstraint =
-          MediaConstraintsHelper::FindBadConstraint(c, aPrefs, mMediaDevice);
-    }
+  if (!ChooseCapability(c, aPrefs, newCapability, distanceMode)) {
+    *aOutBadConstraint =
+        MediaConstraintsHelper::FindBadConstraint(c, aPrefs, mMediaDevice);
     return NS_ERROR_FAILURE;
   }
   LOG("ChooseCapability(%s) for mCapability (Allocate) --",
@@ -479,12 +476,9 @@ nsresult MediaEngineRemoteVideoSource::Reconfigure(
   webrtc::CaptureCapability newCapability;
   LOG("ChooseCapability(%s) for mTargetCapability (Reconfigure) ++",
       ToString(distanceMode));
-  if (!ChooseCapability(c, aPrefs, newCapability, distanceMode,
-                        aOutBadConstraint)) {
-    if (aOutBadConstraint && !*aOutBadConstraint) {
-      *aOutBadConstraint =
-          MediaConstraintsHelper::FindBadConstraint(c, aPrefs, mMediaDevice);
-    }
+  if (!ChooseCapability(c, aPrefs, newCapability, distanceMode)) {
+    *aOutBadConstraint =
+        MediaConstraintsHelper::FindBadConstraint(c, aPrefs, mMediaDevice);
     return NS_ERROR_INVALID_ARG;
   }
   LOG("ChooseCapability(%s) for mTargetCapability (Reconfigure) --",
@@ -893,7 +887,7 @@ static void LogCapability(const char* aHeader,
 bool MediaEngineRemoteVideoSource::ChooseCapability(
     const NormalizedConstraints& aConstraints, const MediaEnginePrefs& aPrefs,
     webrtc::CaptureCapability& aCapability,
-    const DistanceCalculation aCalculate, const char** aOutBadConstraint) {
+    const DistanceCalculation aCalculate) {
   LOG("%s", __PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -913,29 +907,7 @@ bool MediaEngineRemoteVideoSource::ChooseCapability(
     case camera::ScreenEngine:
     case camera::WinEngine:
     case camera::BrowserEngine: {
-      MOZ_ASSERT_IF(aOutBadConstraint, !*aOutBadConstraint);
       FlattenedConstraints c(aConstraints);
-      const auto checkConstraint = [](const auto& aConstraint) {
-        return aConstraint.mMin <= aConstraint.mMax && aConstraint.mMax > 0;
-      };
-      if (!checkConstraint(c.mWidth)) {
-        if (aOutBadConstraint) {
-          *aOutBadConstraint = "width";
-        }
-        return false;
-      }
-      if (!checkConstraint(c.mHeight)) {
-        if (aOutBadConstraint) {
-          *aOutBadConstraint = "height";
-        }
-        return false;
-      }
-      if (!checkConstraint(c.mFrameRate)) {
-        if (aOutBadConstraint) {
-          *aOutBadConstraint = "frameRate";
-        }
-        return false;
-      }
       // DesktopCaptureImpl polls for frames and so must know the framerate to
       // capture at. This is signaled through CamerasParent as the capability's
       // maxFPS. Note that DesktopCaptureImpl does not expose any capabilities.
