diff --git a/dom/ipc/BrowserHost.cpp b/dom/ipc/BrowserHost.cpp
index 6f0aeb1f44..97d68c06c2 100644
--- a/dom/ipc/BrowserHost.cpp
+++ b/dom/ipc/BrowserHost.cpp
@@ -106,7 +106,7 @@ void BrowserHost::UpdateEffects(EffectsInfo aEffects) {
 /* attribute boolean renderLayers; */
 NS_IMETHODIMP
 BrowserHost::GetRenderLayers(bool* aRenderLayers) {
-  if (!mRoot) {
+  if (!mRoot && !Preferences::GetBool("floorp.browser.splitView.working", false)) {
     *aRenderLayers = false;
     return NS_OK;
   }
