diff --git a/layout/style/nsStyleUtil.cpp b/layout/style/nsStyleUtil.cpp
index 027bd7b22d..ab92e79192 100644
--- a/layout/style/nsStyleUtil.cpp
+++ b/layout/style/nsStyleUtil.cpp
@@ -140,15 +140,16 @@ bool nsStyleUtil::ValueIncludes(const nsAString& aValueList,
   return false;
 }
 
-void nsStyleUtil::AppendEscapedCSSString(const nsACString& aString,
-                                         nsACString& aReturn, char aQuoteChar) {
-  MOZ_ASSERT(aQuoteChar == '\'' || aQuoteChar == '"',
+void nsStyleUtil::AppendEscapedCSSString(const nsAString& aString,
+                                         nsAString& aReturn,
+                                         char16_t quoteChar) {
+  MOZ_ASSERT(quoteChar == '\'' || quoteChar == '"',
              "CSS strings must be quoted with ' or \"");
 
-  aReturn.Append(aQuoteChar);
+  aReturn.Append(quoteChar);
 
-  const char* in = aString.BeginReading();
-  const char* const end = aString.EndReading();
+  const char16_t* in = aString.BeginReading();
+  const char16_t* const end = aString.EndReading();
   for (; in != end; in++) {
     if (*in < 0x20 || *in == 0x7F) {
       // Escape U+0000 through U+001F and U+007F numerically.
@@ -159,12 +160,13 @@ void nsStyleUtil::AppendEscapedCSSString(const nsACString& aString,
         // It's not technically necessary to escape the quote
         // character that isn't being used to delimit the string,
         // but we do it anyway because that makes testing simpler.
-        aReturn.Append('\\');
+        aReturn.Append(char16_t('\\'));
       }
       aReturn.Append(*in);
     }
   }
-  aReturn.Append(aQuoteChar);
+
+  aReturn.Append(quoteChar);
 }
 
 /* static */
