diff --git a/mobile/android/fenix/app/src/test/java/org/mozilla/fenix/termsofuse/TermsOfUseManagerTest.kt b/mobile/android/fenix/app/src/test/java/org/mozilla/fenix/termsofuse/TermsOfUseManagerTest.kt
index 24ea6fc5af..3638316197 100644
--- a/mobile/android/fenix/app/src/test/java/org/mozilla/fenix/termsofuse/TermsOfUseManagerTest.kt
+++ b/mobile/android/fenix/app/src/test/java/org/mozilla/fenix/termsofuse/TermsOfUseManagerTest.kt
@@ -5,7 +5,6 @@
 package org.mozilla.fenix.termsofuse
 
 import io.mockk.every
-import io.mockk.spyk
 import junit.framework.TestCase.assertFalse
 import junit.framework.TestCase.assertTrue
 import mozilla.components.support.test.robolectric.testContext
@@ -145,13 +144,12 @@ class TermsOfUseManagerTest {
 
     @Test
     fun `GIVEN other conditions satisfied WHEN prompt has not been displayed THEN shouldShowTermsOfUsePrompt returns true`() {
-        val settings = spyk(settings)
         settings.hasAcceptedTermsOfService = false
         settings.isTermsOfUsePromptEnabled = true
         settings.hasPostponedAcceptingTermsOfUse = false
         // Prompt display count configuration.
         settings.termsOfUsePromptDisplayedCount = 0
-        every { settings.getTermsOfUseMaxDisplayCount() } returns 2
+        settings.termsOfUseMaxDisplayCount = 2
 
         termsOfUseManager.onStart()
 
@@ -160,26 +158,24 @@ class TermsOfUseManagerTest {
 
     @Test
     fun `GIVEN other conditions satisfied WHEN prompt has been displayed the maximum number of times THEN shouldShowTermsOfUsePrompt returns false`() {
-        val settings = spyk(settings)
         settings.hasAcceptedTermsOfService = false
         settings.isTermsOfUsePromptEnabled = true
         settings.hasPostponedAcceptingTermsOfUse = false
         // Prompt display count configuration.
         settings.termsOfUsePromptDisplayedCount = 2
-        every { settings.getTermsOfUseMaxDisplayCount() } returns 2
+        settings.termsOfUseMaxDisplayCount = 2
 
         assertFalse(termsOfUseManager.shouldShowTermsOfUsePrompt())
     }
 
     @Test
     fun `GIVEN other conditions satisfied WHEN prompt has been displayed more than the maximum number of times THEN shouldShowTermsOfUsePrompt returns false`() {
-        val settings = spyk(settings)
         settings.hasAcceptedTermsOfService = false
         settings.isTermsOfUsePromptEnabled = true
         settings.hasPostponedAcceptingTermsOfUse = false
         // Prompt display count configuration.
         settings.termsOfUsePromptDisplayedCount = 3
-        every { settings.getTermsOfUseMaxDisplayCount() } returns 2
+        settings.termsOfUseMaxDisplayCount = 2
 
         assertFalse(termsOfUseManager.shouldShowTermsOfUsePrompt())
     }
