diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/termsofuse/TermsOfUseManager.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/termsofuse/TermsOfUseManager.kt
index 803e5edc6c..a764a4ae38 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/termsofuse/TermsOfUseManager.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/termsofuse/TermsOfUseManager.kt
@@ -53,7 +53,7 @@ class TermsOfUseManager(private val settings: Settings) {
     ): Boolean {
         if (settings.hasAcceptedTermsOfService) return false
         if (!settings.isTermsOfUsePromptEnabled) return false
-        if (settings.termsOfUsePromptDisplayedCount >= settings.getTermsOfUseMaxDisplayCount()) return false
+        if (settings.termsOfUsePromptDisplayedCount >= settings.termsOfUseMaxDisplayCount) return false
 
         val isFirstCheck = isFirstCheckSinceStartingApp
         isFirstCheckSinceStartingApp = false
