diff --git a/taskcluster/gecko_taskgraph/transforms/task.py b/taskcluster/gecko_taskgraph/transforms/task.py
index dcfcf8a94e..084ce0e7e0 100644
--- a/taskcluster/gecko_taskgraph/transforms/task.py
+++ b/taskcluster/gecko_taskgraph/transforms/task.py
@@ -903,7 +903,7 @@ def build_iscript_payload(config, task, task_def):
     "beetmover",
     schema={
         # the maximum time to run, in seconds
-        Optional("max-run-time"): int,
+        Required("max-run-time"): int,
         # locale key, if this is a locale beetmover job
         Optional("locale"): str,
         Required("release-properties"): {
@@ -959,7 +959,7 @@ def build_beetmover_payload(config, task, task_def):
     "beetmover-push-to-release",
     schema={
         # the maximum time to run, in seconds
-        Optional("max-run-time"): int,
+        Required("max-run-time"): int,
         Required("product"): str,
     },
 )
@@ -979,7 +979,7 @@ def build_beetmover_push_to_release_payload(config, task, task_def):
 @payload_builder(
     "beetmover-import-from-gcs-to-artifact-registry",
     schema={
-        Optional("max-run-time"): int,
+        Required("max-run-time"): int,
         Required("gcs-sources"): [str],
         Required("product"): str,
     },
@@ -994,7 +994,7 @@ def build_import_from_gcs_to_artifact_registry_payload(config, task, task_def):
 @payload_builder(
     "beetmover-maven",
     schema={
-        Optional("max-run-time"): int,
+        Required("max-run-time"): int,
         Required("release-properties"): {
             "app-name": str,
             "app-version": str,
@@ -1769,7 +1769,14 @@ def set_defaults(config, tasks):
                     "Windows and Linux, not on {}".format(worker["os"])
                 )
             worker.setdefault("chain-of-trust", False)
-        elif worker["implementation"] in ("iscript",):
+        elif worker["implementation"] in (
+            "beetmover",
+            "beetmover-push-to-release",
+            "beetmover-maven",
+            "beetmover-import-from-gcs-to-artifact-registry",
+            "iscript",
+            "scriptworker-signing",
+        ):
             worker.setdefault("max-run-time", 600)
         elif worker["implementation"] == "push-apk":
             worker.setdefault("commit", False)
