diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 2512c029ec8..d55ef71a100 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -273,6 +273,18 @@
 #  include "DBusService.h"
 #endif
 
+/*
+ * NORANEKO PATCH
+ * [updater]
+ * START
+ */
+#include "mozilla/URLPreloader.h"
+/*
+ * NORANEKO PATCH
+ * [updater]
+ * END
+ */
+
 extern void InstallSignalHandlers(const char* ProgramName);
 
 #define FILE_COMPATIBILITY_INFO "compatibility.ini"_ns
@@ -3939,6 +3951,51 @@ CompatCheckResult CheckCompatibility(nsIFile* aProfileDir,
 
   if (result.isDifferentInstall) return result;
 
+  /*
+   * NORANEKO PATCH
+   * [updater]
+   * START
+   */
+  {
+    auto _NRReadString = [](nsIFile* aFile, nsACString& buildid) -> nsresult {
+      nsCString tmp = MOZ_TRY(URLPreloader::ReadFile(aFile));
+      buildid = tmp;
+      return NS_OK;
+    };
+    nsCOMPtr<nsIFile> buildid2_profile;
+    aProfileDir->Clone(getter_AddRefs(buildid2_profile));
+    if (!buildid2_profile) return result;
+    buildid2_profile->AppendNative("buildid2"_ns);
+
+    nsCString buildid2_profile_string;
+    rv = _NRReadString(buildid2_profile, buildid2_profile_string);
+    if (NS_FAILED(rv)) {
+      return result;
+    }
+
+    if (aAppDir) {
+      nsCOMPtr<nsIFile> buildid2_appDir;
+      aAppDir->Clone(getter_AddRefs(buildid2_appDir));
+      if (!buildid2_appDir) return result;
+      buildid2_appDir->AppendNative("buildid2"_ns);
+
+      nsCString buildid2_appDir_string;
+      rv = _NRReadString(buildid2_appDir, buildid2_appDir_string);
+      if (NS_FAILED(rv)) {
+        return result;
+      }
+
+      if (!buildid2_profile_string.Equals(buildid2_appDir_string)) {
+        return result;
+      }
+    }
+  }
+  /*
+   * NORANEKO PATCH
+   * [updater]
+   * END
+   */
+
   // If we see this flag, caches are invalid.
   rv = parser.GetString("Compatibility", "InvalidateCaches", buf);
   result.cachesOK = (NS_FAILED(rv) || !buf.EqualsLiteral("1"));
