diff --git a/toolkit/crashreporter/test/unit/test_crashreporter_crash.js b/toolkit/crashreporter/test/unit/test_crashreporter_crash.js
index 0d95ef0387..e4f89a7833 100644
--- a/toolkit/crashreporter/test/unit/test_crashreporter_crash.js
+++ b/toolkit/crashreporter/test/unit/test_crashreporter_crash.js
@@ -113,7 +113,6 @@ add_task(async function run_test() {
       // Enable the FHR, official policy bypass (since we're in a test) and
       // specify a telemetry server & client ID.
       Services.prefs.setBoolPref("termsofuse.bypassNotification", true);
-      Services.prefs.setBoolPref("browser.preonboarding.enabled", false);
       Services.prefs.setBoolPref(
         "datareporting.policy.dataSubmissionPolicyBypassNotification",
         true
