diff --git a/widget/cocoa/nsCocoaFeatures.mm b/widget/cocoa/nsCocoaFeatures.mm
index 008876051b..509bdbd3fb 100644
--- a/widget/cocoa/nsCocoaFeatures.mm
+++ b/widget/cocoa/nsCocoaFeatures.mm
@@ -27,19 +27,16 @@
 #define MACOS_VERSION_12_0_HEX 0x000C0000
 #define MACOS_VERSION_13_0_HEX 0x000D0000
 #define MACOS_VERSION_14_0_HEX 0x000E0000
-#define MACOS_VERSION_26_0_HEX 0x001A0000
 
 #include "nsCocoaFeatures.h"
 #include "nsCocoaUtils.h"
 #include "nsDebug.h"
 #include "nsObjCExceptions.h"
 
-#include <atomic>
 #import <Cocoa/Cocoa.h>
 #include <sys/sysctl.h>
 
-// The lazily-initialized version. 0 before initialization, non-zero after.
-static std::atomic<int32_t> sOSVersion = 0;
+/*static*/ int32_t nsCocoaFeatures::mOSVersion = 0;
 
 // This should not be called with unchecked aMajor, which should be >= 10.
 inline int32_t AssembleVersion(int32_t aMajor, int32_t aMinor,
@@ -119,8 +116,8 @@ int32_t nsCocoaFeatures::GetVersion(int32_t aMajor, int32_t aMinor,
   return macOSVersion;
 }
 
-/*static*/ int32_t nsCocoaFeatures::ComputeVersion() {
-  NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
+/*static*/ void nsCocoaFeatures::InitializeVersionNumbers() {
+  NS_OBJC_BEGIN_TRY_IGNORE_BLOCK;
 
   // Provide an autorelease pool to avoid leaking Cocoa objects,
   // as this gets called before the main autorelease pool is in place.
@@ -128,25 +125,19 @@ int32_t nsCocoaFeatures::GetVersion(int32_t aMajor, int32_t aMinor,
 
   int major, minor, bugfix;
   GetSystemVersion(major, minor, bugfix);
-  return GetVersion(major, minor, bugfix);
+  mOSVersion = GetVersion(major, minor, bugfix);
 
-  NS_OBJC_END_TRY_BLOCK_RETURN(0);
+  NS_OBJC_END_TRY_IGNORE_BLOCK;
 }
 
 /* static */ int32_t nsCocoaFeatures::macOSVersion() {
-  int32_t version = sOSVersion.load(std::memory_order_relaxed);
-  if (version != 0) {
-    return version;
+  // Don't let this be called while we're first setting the value...
+  MOZ_ASSERT((mOSVersion & MACOS_VERSION_MASK) >= 0);
+  if (!mOSVersion) {
+    mOSVersion = -1;
+    InitializeVersionNumbers();
   }
-
-  // Compute the version. Multiple threads might do the computation
-  // concurrently. That's ok; they will all compute the same value.
-  version = ComputeVersion();
-  if (version != 0) {
-    sOSVersion.store(version, std::memory_order_relaxed);
-  }
-
-  return version;
+  return mOSVersion;
 }
 
 /* static */ int32_t nsCocoaFeatures::macOSVersionMajor() {
@@ -195,11 +186,6 @@ int32_t nsCocoaFeatures::GetVersion(int32_t aMajor, int32_t aMinor,
   return (macOSVersion() >= MACOS_VERSION_14_0_HEX);
 }
 
-/* static */ bool nsCocoaFeatures::OnTahoeOrLater() {
-  // See comments above regarding SYSTEM_VERSION_COMPAT.
-  return (macOSVersion() >= MACOS_VERSION_26_0_HEX);
-}
-
 /* static */ bool nsCocoaFeatures::IsAtLeastVersion(int32_t aMajor,
                                                     int32_t aMinor,
                                                     int32_t aBugFix) {
