diff --git a/widget/cocoa/nsMenuX.h b/widget/cocoa/nsMenuX.h
index 3fd27753e0..cdc5884a99 100644
--- a/widget/cocoa/nsMenuX.h
+++ b/widget/cocoa/nsMenuX.h
@@ -175,13 +175,11 @@ class nsMenuX final : public nsMenuParentX,
   // nsMenuParentX
   void MenuChildChangedVisibility(const MenuChild& aChild,
                                   bool aIsVisible) override;
-  size_t NestingDepth() override { return mNestingDepth; }
 
   void Dump(uint32_t aIndent) const;
 
   static bool IsXULHelpMenu(nsIContent* aMenuContent);
   static bool IsXULWindowMenu(nsIContent* aMenuContent);
-  static bool IsXULEditMenu(nsIContent* aMenuContent);
 
   // Set an observer that gets notified of menu opening and closing.
   // The menu does not keep a strong reference the observer. The observer must
@@ -293,8 +291,6 @@ class nsMenuX final : public nsMenuParentX,
   // items.
   mozilla::Maybe<uint32_t> mHighlightedItemIndex;
 
-  size_t mNestingDepth = 0;
-
   bool mIsEnabled = true;
   bool mNeedsRebuild = true;
 
