diff --git a/widget/cocoa/nsMenuBarX.mm b/widget/cocoa/nsMenuBarX.mm
index 35f8698b00..9ce72f0ecc 100644
--- a/widget/cocoa/nsMenuBarX.mm
+++ b/widget/cocoa/nsMenuBarX.mm
@@ -492,15 +492,7 @@ nsresult nsMenuBarX::Paint() {
   NS_OBJC_END_TRY_ABORT_BLOCK;
   NS_OBJC_BEGIN_TRY_ABORT_BLOCK;
 
-  // If the user switches to another app and back to the last open window, we
-  // should not remove the problematic menu items again or the emoji picker
-  // would not be able to be opened again via shortcuts. This should be the only
-  // time that `sLastGeckoMenuBarPainted` is checked in this method, since other
-  // optimizations could interfere with menu manipulations by native dialogs and
-  // similar (see comment above).
-  if (nsMenuBarX::sLastGeckoMenuBarPainted != this) {
-    RemoveProblematicMenuItems(mNativeMenu);
-  }
+  RemoveProblematicMenuItems(mNativeMenu);
 
   NS_OBJC_END_TRY_ABORT_BLOCK;
   NS_OBJC_BEGIN_TRY_ABORT_BLOCK;
