diff --git a/xpcom/threads/nsThread.cpp b/xpcom/threads/nsThread.cpp
index 73bdd99166..1cc97cbf92 100644
--- a/xpcom/threads/nsThread.cpp
+++ b/xpcom/threads/nsThread.cpp
@@ -86,8 +86,6 @@ using GetCurrentThreadStackLimitsFn = void(WINAPI*)(PULONG_PTR LowLimit,
 #  include <mach/mach.h>
 #  include <mach/thread_policy.h>
 #  include <sys/qos.h>
-
-#  include "nsCocoaFeatures.h"
 #endif
 
 #ifdef MOZ_CANARY
@@ -340,14 +338,6 @@ void nsThread::ThreadFunc(void* aArg) {
 
   self->InitCommon();
 
-#ifdef XP_MACOSX
-  if (nsCocoaFeatures::OnTahoeOrLater()) {
-    // On macOS 26+, use "User Initiated" as the default quality of service.
-    // It may make sense to do this on all versions of macOS.
-    pthread_set_qos_class_self_np(QOS_CLASS_USER_INITIATED, 0);
-  }
-#endif
-
   // Inform the ThreadManager
   nsThreadManager::get().RegisterCurrentThread(*self);
 
